src/overlays/fbdemos/ovl_fbdemo_wipe4/z_fbdemo_wipe4.c

```diff
@@ -16,35 +26,115 @@ void TransitionWipe4_SetColor(void* thisx, u32 color);
 void TransitionWipe4_SetEnvColor(void* thisx, u32 color);
 s32 TransitionWipe4_IsDone(void* thisx);
 
-#if 0
 TransitionInit TransitionWipe4_InitVars = {
-    TransitionWipe4_Init,
-    TransitionWipe4_Destroy,
-    TransitionWipe4_Update,
-    TransitionWipe4_Draw,
-    TransitionWipe4_Start,
-    TransitionWipe4_SetType,
-    TransitionWipe4_SetColor,
-    TransitionWipe4_SetEnvColor,
-    TransitionWipe4_IsDone,
+    TransitionWipe4_Init,     TransitionWipe4_Destroy,     TransitionWipe4_Update,
+    TransitionWipe4_Draw,     TransitionWipe4_Start,       TransitionWipe4_SetType,
+    TransitionWipe4_SetColor, TransitionWipe4_SetEnvColor, TransitionWipe4_IsDone,
 };
 
-#endif
+void TransitionWipe4_Start(void* thisx) {
+    TransitionWipe4* this = (TransitionWipe4*)thisx;
+
+    this->isDone = false;
+
+    switch (this->speedType) {
+        default:
+            this->baseSpeed = 0.2f;
+            break;
+
+        case 0:
+            this->baseSpeed = 0.1f;
+            break;
+
+        case 1:
+            this->baseSpeed = 0.05f;
+            break;
+    }
+
+    switch (this->colorType) {
+        default:
+            this->primColor.rgba = RGBA8(160, 160, 160, 255);
+            break;
+
+        case 0:
+            this->primColor.rgba = RGBA8(0, 0, 0, 255);
+            break;
+
+        case 1:
+            this->primColor.rgba = RGBA8(160, 160, 160, 255);
+            break;
+    }
+}
+
+void* TransitionWipe4_Init(void* thisx) {
+    TransitionWipe4* this = (TransitionWipe4*)thisx;
+
+    bzero(this, sizeof(TransitionWipe4));
+
+    return this;
+}
+
+void TransitionWipe4_Destroy(void* thisx) {
+}
+
+void TransitionWipe4_Update(void* thisx, s32 updateRate) {
+    TransitionWipe4* this = (TransitionWipe4*)thisx;
+
+    this->progress += (this->baseSpeed * 3.0f) / updateRate;
+    if (this->progress >= 1.0f) {
+        this->progress = 1.0f;
+        this->isDone = true;
+    }
+}
+
+// Use of THIS in this function is required to match
+void TransitionWipe4_Draw(void* thisx, Gfx** gfxP) {
+    Gfx* gfx = *gfxP;
+    Struct_80140E80* var_a3 = &THIS->bg;
```

Comment:
```suggestion
    Struct_80140E80* bg = &THIS->bg;
```

Reply:
For now, but obviously this will change when the system is properly named.